packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -109,7 +110,8 @@ class UrlLauncherPlugin extends UrlLauncherPlatform {
   @override
   Future<bool> launchUrl(String url, LaunchOptions options) async {
     final String? windowName = options.webOnlyWindowName;
-    return openNewWindow(url, webOnlyWindowName: windowName) != null;
+    openNewWindow(url, webOnlyWindowName: windowName);
+    return canLaunch(url);
```

Comment:
This should just `return true;`, no need to call `canLaunch` or anything else.

(Question: if `canLaunch` returns false, does it prevent `launchUrl` from actually attempting to open a browser tab with a bogus URL? I don't think it would, right?)

Reply:
Returning `true` will break the [url_launcher_web_test test](https://github.com/Frank3K/flutter_packages/blob/0c29f9463acb09a5ad8f1ada9ff16e7b8221bf68/packages/url_launcher/url_launcher_web/example/integration_test/url_launcher_web_test.dart#L119-L122).

I'll update the code and the test.

> Question: if canLaunch returns false, does it prevent launchUrl from actually attempting to open a browser tab with a bogus URL? I don't think it would, right?

In the current setup it won't. If we would want that, we could do a `canLaunch` earlier and bail-out. But I agree returning true is better; if the user of this package wants such behaviour he can do so himself by calling `canLaunch` first in his own code.

Reply:
`canLaunch` is not super valuable on the web, the browser doesn't know if it can handle a protocol and will just attempt to open anything, I think

Reply:
@Frank3K how about:

* `html.Window? openNewWindow(String url, {String? webOnlyWindowName})` becomes:
  * `bool openNewWindow(String url, {String? webOnlyWindowName})`

If it does NOT call `window.open` -> `return false`
If it calls `window.open` -> `return true`

Then the implementation of `launchUrl` can be just:

`return openNewWindow...`

That way we can still return `false` if the user really tries to mess up (like opening a **disallowed** protocol).

(I can incorporate those changes to this PR myself, if you think they make sense)

Reply:
Good idea 👍 . I applied the proposed changes and added a small test.

This PR is ready for re-review.